### PR TITLE
[Collecticiel]Collecticiel version for 16 05 1 milestone

### PR DESCRIPTION
**Pull request by ericvincenterv**
Some buges fixes and some modifications for "Collecticiel 16-05-1".

### CODE REVIEW

**Comment by LaurentGruber** [comment]
LGTM

File: plugin/collecticiel/Controller/DropController.php [hunk 1]
```diff
@@ -457,6 +477,11 @@ public function dropsAwaitingAction(Dropzone $dropzone, $page)
             $dropUserId = $drop->getUser()->getId();
             $userToCommentCount[$dropUserId] = $nbCommentsPerUser;
             $userNbTextToRead[$dropUserId] = $nbTextToRead;
+
+            $notationCommentDocuments = $dropManager->getNotationCommentForDocuments($drop);
+            $notationQualityDocuments = $dropManager->getNotationQualityForDocuments($drop);
+            $notationAssessorDocuments = $dropManager->getNotationAssessorForDocuments($drop);
+            $notationAppreciationDocuments = $dropManager->getAppreciationForDocuments($drop);
```
Comment (Elorfin) [inline plugin/collecticiel/Controller/DropController.php]: Seems like you override the content of this 4 arrays (so at the end of the loop you only have the results for the last `$drop`).
It depends on what you want to do, but you need either to move this methods outside the loop or merge the results of the functions with the previous content of the array.

File: plugin/collecticiel/Controller/DropzoneController.php [hunk 2]
```diff
@@ -85,6 +85,9 @@ public function editCommonAction(Dropzone $dropzone, $user)
             if ($dropzone->getEvaluationType() != Dropzone::EVALUATION_TYPE) {
                 $dropzone->setEvaluation(1);
             }
+            if ($dropzone->getEvaluationType() == Dropzone::EVALUATION_TYPE) {
```
Comment (Elorfin) [inline plugin/collecticiel/Controller/DropzoneController.php]: Use strict comparison `===`.

File: plugin/collecticiel/Controller/NotationController.php [hunk 3]
```diff
@@ -47,18 +47,117 @@ public function AddNotationForDocsAction()
...
+
+        if ($recordOrTransmit == 0) {
+            // Ajout pour avoir si la notation a été transmise ou pas.
+            $notation = $em->getRepository('InnovaCollecticielBundle:Notation')
+                        ->findBy(
+                                array(
+                                    'document' => $document->getId(),
+                                    'dropzone' => $dropzone->getId(),
+                                     )
+                                );
```
Comment (Elorfin) [inline plugin/collecticiel/Controller/NotationController.php]: This block can be moved outside the `if` statement : also called in the `else` part.

File: plugin/collecticiel/Controller/NotationController.php [hunk 4]
```diff
@@ -108,12 +209,27 @@ public function ajaxValidateTransmitEvaluationDocumentAction(Document $document,
         $em->persist($notation[0]);
         $em->flush();
 
+        $gradingScale = $em->getRepository('InnovaCollecticielBundle:GradingScale')
+                      ->find($notation[0]->getAppreciation());
```
Comment (Elorfin) [inline plugin/collecticiel/Controller/NotationController.php]: `$notation[0]` may not exist.

File: plugin/collecticiel/Manager/GradingCriteriaManager.php [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+     *  To update gradingScale table.
+     *
+     * @param tab
+     *
+     * @return bool
+     */
+    public function manageGradingCriterias($tab, Dropzone $dropzone)
+    {
+        // handle old scales deletion
+        $this->deletOldCriterias($tab, $dropzone);
```
Comment (Elorfin) [inline plugin/collecticiel/Manager/GradingCriteriaManager.php]: Typo in function name.

File: plugin/collecticiel/Manager/GradingCriteriaManager.php [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        // handle update and add
+        foreach (array_keys($tab) as $key) {
```
Comment (Elorfin) [inline plugin/collecticiel/Manager/GradingCriteriaManager.php]: Do not call function here : assign the result of `array_keys` to a variable and loop through it.

File: plugin/collecticiel/Resources/public/js/popup.js [hunk 7]
```diff
@@ -489,38 +522,57 @@ $(document).ready(function() {
...
+        // Récupération de l'id du document
+        var documentId = $(this).attr("data-document_id");
 
         // Récupération de l'id du document
-        var note = document.getElementById('innova_collecticiel_notation_form_note').value;
-        var commentText = document.getElementById('innova_collecticiel_notation_form_commentText').value;
-        var qualityText = document.getElementById('innova_collecticiel_notation_form_qualityText').value;
+        var evaluationType = $(this).attr("data_document_evaluationType");
+
+        if (evaluationType == 'notation') {
```
Comment (Elorfin) [inline plugin/collecticiel/Resources/public/js/popup.js]: Use strict comparison

File: plugin/collecticiel/Resources/public/js/popup.js [hunk 8]
```diff
@@ -489,38 +522,57 @@ $(document).ready(function() {
...
+            var appreciation = 0;
+            var commentText = "";
+            var qualityText = "";
+            var note = document.getElementById('innova_collecticiel_notation_form_note_'+documentId).value;
+        }
+
+        if (evaluationType == 'ratingScale') {
```
Comment (Elorfin) [inline plugin/collecticiel/Resources/public/js/popup.js]: Use strict comparison

File: plugin/collecticiel/Controller/DropzoneController.php (hunk 2 above)
Comment (ericvincenterv) [inline plugin/collecticiel/Controller/DropzoneController.php]: ok

File: plugin/collecticiel/Controller/NotationController.php (hunk 3 above)
Comment (ericvincenterv) [inline plugin/collecticiel/Controller/NotationController.php]: ok

File: plugin/collecticiel/Controller/NotationController.php (hunk 4 above)
Comment (ericvincenterv) [inline plugin/collecticiel/Controller/NotationController.php]: $notation[0] is declared 15 lines before.

File: plugin/collecticiel/Resources/public/js/popup.js (hunk 7 above)
Comment (ericvincenterv) [inline plugin/collecticiel/Resources/public/js/popup.js]: ok

File: plugin/collecticiel/Resources/public/js/popup.js (hunk 8 above)
Comment (ericvincenterv) [inline plugin/collecticiel/Resources/public/js/popup.js]: ok

File: plugin/collecticiel/Manager/GradingCriteriaManager.php (hunk 5 above)
Comment (ericvincenterv) [inline plugin/collecticiel/Manager/GradingCriteriaManager.php]: ok

File: plugin/collecticiel/Manager/GradingCriteriaManager.php (hunk 6 above)
Comment (ericvincenterv) [inline plugin/collecticiel/Manager/GradingCriteriaManager.php]: ok

File: plugin/collecticiel/Controller/DropController.php (hunk 1 above)
Comment (ericvincenterv) [inline plugin/collecticiel/Controller/DropController.php]: Yes, I add an issue for the next release, I didn't have time to change now and test for Wednesday : https://github.com/claroline/Distribution/issues/692

**Comment by Elorfin** [comment]
LGTM

**Pull request merged**